### PR TITLE
feat: retrieve top rated testimonials

### LINKED ISSUE #1142: [FEAT]: Fetch Top-Rated Testimonials
### Description:
This feature would allow users to fetch testimonials with the highest ratings and this would increase the visibility for top-rated feedback.

### Endpoint Details:

- **Method:** GET  
- **Endpoint:** `/api/v1/testimonials/top-rated`  
- **Description:** Fetches a list of top-rated testimonials, sorted by rating in descending order.

### Request Example:

```sh
GET "/api/v1/testimonials/top-rated"
```

#### **Success Response:**
```json
{
  "status": "success",
  "status_code": 200,
  "message": "Top-rated testimonials retrieved successfully.",
  "data": [
    {
      "id": "1",
      "content": "Great service!",
      "ratings": 5,
      "author_id": "123",
      "created_at": "2024-02-28T10:00:00Z"
    },
    {
      "id": "2",
      "content": "Amazing experience!",
      "ratings": 5,
      "author_id": "456",
      "created_at": "2024-02-27T15:30:00Z"
    }
  ]
}
```

#### **No Testimonials Found:**
```json
{
  "status": "success",
  "status_code": 200,
  "message": "No testimonials found.",
  "data": []
}
```

#### **Server Error:**
```json
{
  "status": "error",
   "status_code": 500,
  "message": "An unexpected error occurred.",
  "data": null
}
```

### Tasks:

- Add a `fetch_top_rated` method in `TestimonialService` that:
  - Queries testimonials ordered by rating (descending order).
  - Supports pagination (`page`, `per_page`).
- Handle edge cases such as an empty testimonial list.

### Testing:

Write unit tests for:

1. Fetching top-rated testimonials with valid data.
2. Handling cases where no testimonials exist.
3. Validating pagination parameters.

### PR DESCRIPTION
**Pull request by mikemando**
## Description
This PR introduces a new API endpoint `/api/v1/testimonials/top-rated` that allows users to retrieve testimonials sorted by rating. It also includes a unit test to ensure the API works as expected. 

​

## Related Issue (Link to issue ticket)
closes #1142

​

## Motivation and Context
- Provides users with easy access to the highest-rated testimonials.  
- Supports pagination to improve performance and usability.   

​

## How Has This Been Tested?
- **Unit Testing**:  
  - Implemented a test (`test_get_top_rated_api`) using `pytest` and `unittest.mock.MagicMock`.  
  - Validates that the API returns a **200 OK** response.  
  - Confirms the response structure, message, and data.  
  - Tests pagination functionality.  

​

## Screenshots (if appropriate - Postman, etc):
top_rated_endpoint

​

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
      ​

## Checklist:

- [x] My code follows the code style of this project.
- [x] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the **CONTRIBUTING** document.
- [x] I have added tests to cover my changes.
- [x] All new and existing tests passed.

### CODE REVIEW

File: api/v1/services/testimonial.py [hunk 1]
```diff
@@ -54,6 +56,25 @@ def delete_all(self, db: Session):
...
             db.rollback()
             raise e
+    
+    def top_rated_testimonials(self, db: Session, page: int = 1, per_page: int = 10):
+        """
+        Fetch testimonials with the highest ratings and paginates the results.
+        """
+        if page < 1 or per_page < 1:
+            raise HTTPException(status_code=status.HTTP_400_BAD_REQUEST, detail="Invalid pagination parameters")
+
```
Comment (samuelogboye) [inline api/v1/services/testimonial.py]: Instead of manually checking page < 1 or per_page < 1, FastAPI's Query parameters already enforce the constraints (ge=1), so this check is redundant.

File: tests/v1/testimonial/test_top_rated_testimonial.py [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+    mock_testimonial.author_id = "abc"
+    mock_testimonial.created_at = datetime.now()
+
+    db_session_mock.query().order_by().offset().limit().all.return_value = [mock_testimonial]
+    
+    response = client.get("/api/v1/testimonials/top-rated", params={"page": 1, "per_page": 1})
+    
+    assert response.status_code == 200
+    assert response.json()["message"] == "Top-rated testimonials retrieved successfully."
+    assert len(response.json()["data"]) == 1
```
Comment (samuelogboye) [inline tests/v1/testimonial/test_top_rated_testimonial.py]: Add tests for edge cases like
No testimonials in the database, Invalid page or per_page parameters, Database failure scenario.

**Review by samuelogboye — DISMISSED** [review]
LGTM
